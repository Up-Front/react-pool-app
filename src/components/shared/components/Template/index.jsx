import React from 'react';
import styled, { keyframes } from 'styled-components'
import { BrowserRouter as Router, Link } from 'react-router-dom';


const Wrapper = styled.section`
  background: papayawhip;
  display: flex;
  align-items: center;
  height:100%;
  flex-direction:column;
  padding:0;
`;

const Segment = styled.section`
  padding: 10px;
  background:white;
  display: flex;
  justify-content: ${props => props.spaceBetween ? 'space-between' : 'center'};
  ${props => props.selfStart ? 'align-self: flex-start;' : ''}
  align-items: center;
  flex-direction:${props => props.row ? 'row' : 'column'};
  border-radius:8px;
  width:100%;
  box-sizing: border-box;
`;

const FullScreen = styled.section`
flex: 1;
display: flex;
justify-content: center;
align-items: center;
`;

const Template = (props) => (
    <Wrapper>
        <Segment>
            <Link to='/'>/</Link>
            <Link to='/login'>login</Link>
            <Link to='/dashboard'>dashboard</Link>
            <Link to='/leaderboard'>leaderboard</Link>
        </Segment>
        <FullScreen>
            {props.children}
        </FullScreen>
    </Wrapper>
);

export default Template;