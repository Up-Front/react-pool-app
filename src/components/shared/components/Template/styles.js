import styled from 'styled-components';

export const Wrapper = styled.section`
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 100%;
  width: 100%;
  padding: 0;
  background: ${props => props.theme.background};
  background-image: ${props => props.theme.backgroundImage};
  font-size: ${props => props.theme.fontSize};

  & * {
    box-sizing: border-box;
  }
`;

export const Header = styled.header`
  display: flex;
  justify-content: 'space-between';
  ${props => (props.selfStart ? 'align-self: flex-start;' : '')};
  align-items: center;
  flex-direction: 'row';
  padding: 10px;
  width: 100%;
  height: 50px;
  background-color: ${props => props.theme.headerBackgroundColor};

  & a {
    color: ${props => props.theme.headerFontColor};
  }
`;

export const HeaderLogo = styled.h1``;

export const FullScreen = styled.section`
  flex: 1;
  width: 100%;
  margin: 5px 0;
  overflow-y: scroll;
`;
