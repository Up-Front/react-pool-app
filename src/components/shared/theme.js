// this is the theme for styled components colors
export const theme = {
  fontColor: '#FFF',
  fontSize: '16px',
  background: '#243949',
  borderColor: '#FFF',
  backgroundImage: 'linear-gradient(to top, #243949 70%, #517fa4 100%)',

  headerBackgroundColor: 'rgba(0, 0, 0, 0.7)',
  headerFontColor: '#FFF',

  buttonColor: 'DarkOrange',
  buttonFontColor: '#FFF',

  //match
  matchBackgroundColor: '#243949',
  contestedMatchBackgroundColor: 'red',

  //delete swipe
  deleteBackgroundColor: 'red',

  //user / competitor
  statusOnlineColor: 'green',
  statusOfflineColor: 'orange',
  statusUnknownColor: 'lightgrey',
};
