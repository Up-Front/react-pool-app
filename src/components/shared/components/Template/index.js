import React from 'react';
import { Link } from 'react-router-dom';
import { Wrapper, Header, FullScreen, HeaderLogo } from './styles';
import { logoutUser } from './../../../../actions/auth';

const Template = props => (
  <Wrapper>
    <Header>
      <Link to="/">
        <HeaderLogo role="banner">
          FrontMen
        </HeaderLogo>
      </Link>
      <button onClick={logoutUser}>logout</button>
      <Link to="/leaderboard">leaderboard</Link>
    </Header>
    <FullScreen>{props.children}</FullScreen>
  </Wrapper>
);

export default Template;
