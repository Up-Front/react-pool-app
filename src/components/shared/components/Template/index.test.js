import React from 'react';
import { shallow } from 'enzyme';
import theme from './../../theme';
import Template from './';

describe('Template Component', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<Template them={theme} />);
    expect(wrapper).toMatchSnapshot();
  });
});
