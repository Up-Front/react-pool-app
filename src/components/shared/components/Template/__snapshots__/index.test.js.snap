// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template Component renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Template
    them={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.header>
          <Link
                    replace={false}
                    to="/"
          >
                    <styled.h1
                              role="banner"
                    >
                              FrontMen
                    </styled.h1>
          </Link>
          <button
                    onClick={[Function]}
          >
                    logout
          </button>
          <Link
                    replace={false}
                    to="/leaderboard"
          >
                    leaderboard
          </Link>
</styled.header>,
        <styled.section>
          <Switch>
                    <Route
                              component={[Function]}
                              path="/leaderboard"
                    />
                    <Route
                              component={[Function]}
                              path="/"
                    />
          </Switch>
</styled.section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
>
              <styled.h1
                            role="banner"
              >
                            FrontMen
              </styled.h1>
</Link>,
            <button
              onClick={[Function]}
>
              logout
</button>,
            <Link
              replace={false}
              to="/leaderboard"
>
              leaderboard
</Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styled.h1
                role="banner"
>
                FrontMen
</styled.h1>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "FrontMen",
                "role": "banner",
              },
              "ref": null,
              "rendered": "FrontMen",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "logout",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "logout",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "leaderboard",
              "replace": false,
              "to": "/leaderboard",
            },
            "ref": null,
            "rendered": "leaderboard",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        path="/leaderboard"
            />
            <Route
                        component={[Function]}
                        path="/"
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                path="/leaderboard"
/>,
              <Route
                component={[Function]}
                path="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/leaderboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.header>
            <Link
                        replace={false}
                        to="/"
            >
                        <styled.h1
                                    role="banner"
                        >
                                    FrontMen
                        </styled.h1>
            </Link>
            <button
                        onClick={[Function]}
            >
                        logout
            </button>
            <Link
                        replace={false}
                        to="/leaderboard"
            >
                        leaderboard
            </Link>
</styled.header>,
          <styled.section>
            <Switch>
                        <Route
                                    component={[Function]}
                                    path="/leaderboard"
                        />
                        <Route
                                    component={[Function]}
                                    path="/"
                        />
            </Switch>
</styled.section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
>
                <styled.h1
                                role="banner"
                >
                                FrontMen
                </styled.h1>
</Link>,
              <button
                onClick={[Function]}
>
                logout
</button>,
              <Link
                replace={false}
                to="/leaderboard"
>
                leaderboard
</Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <styled.h1
                  role="banner"
>
                  FrontMen
</styled.h1>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "FrontMen",
                  "role": "banner",
                },
                "ref": null,
                "rendered": "FrontMen",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "logout",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "logout",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "leaderboard",
                "replace": false,
                "to": "/leaderboard",
              },
              "ref": null,
              "rendered": "leaderboard",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            path="/leaderboard"
              />
              <Route
                            component={[Function]}
                            path="/"
              />
</Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  path="/leaderboard"
/>,
                <Route
                  component={[Function]}
                  path="/"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/leaderboard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
